src/lib.rs

```diff
@@ -156,12 +165,53 @@ impl<T> Deref for Gc<T> {
     }
 }
 
+pub fn init(flags: gc::DebugFlags) {
+    unsafe { COLLECTOR = Some(Collector::new(flags)) };
+}
+
+pub fn collect() {
+    // FIXME: Is this UB? Theoretically this pub `collect` fn can be called in
+    // parallel from multiple threads, allowing us potentially multiple mutable
+    // references to the underlying `COLLECTOR` struct. However, the inner
+    // `collect` method guards against multiple collection invocations via a
```

Comment:
Why would this be UB? Merely having a reference to a struct in multiple threads cannot lead to UB (unless the underlying memory moves or similar!): it's reading and writing that one has to be careful about.

Reply:
Makes sense. I've removed the comment.

Reply:
As long as there are no aliasing problems... I have no idea.